wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/spi/NamespaceResolutionContext.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.workspace.server.spi;
+
+import java.util.Objects;
+
+/**
+ * Infrastructure may be configured to use namespace value that should be evaluated. This objects
+ * holds everything that might be needed for such evaluation.
```

Comment:
I would lead with what the class is for ("Holds information needed for namespace name evaluation") and follow up with why in what scenarios it's needed.